plugins/convert_README_to_header.sh

```diff
@@ -30,8 +30,8 @@ outfilename=`basename $outfile`;
 # See http://stackoverflow.com/a/16576291
 # On Mac OS's sed, \n is not recognized as a newline character, but
 # \[actual newline] works
-desc=`sed -e ':a' -e 'N' -e '$!ba' -e 's/\"/\\\"/g' -e 's/\n/\\\\n"\\
-"/g' "$path/README.md"`;
+desc=`sed -e ':a' -e 'N' -e '$!ba' -e 's/\\\/\\\\\\\/g' -e 's/\"/\\\"/g' -e 's/\n/\\\\n"\\
+"#g' "$path/README.md"`;
```

Comment:
This last hash needs fixing too. I'm not quite sure why Travis isn't picking it up.

Reply:
As you can see I moved the whole regex to next line that will prevent such oversights in the future.

Reply:
You've actually just broken it on OS X. Please see the comment a line or two above in the code:
https://github.com/OpenLightingProject/ola/pull/1586/files#diff-e27975dbf242f99283db8699d57acd1cL30-L32

Reply:
On Linux we are also not inserting newlines.... we're inserting "\n" into a C++ string that is actually one of the reasons lint doesn't like these files....
For a "true" newline to be insterted it needs to not be escaped....

Reply:
This goes back to should we care whether or not autogenerated content goes over 80 chars, if we don't we don't need to insert "true" newlines and just adding "\n" is sufficient for the C++.

If we do then it needs to be dine right all the way on both Linux and OS X.

To me option 1 is the more convincing one, we're "reading" these files only through the C++ code that is rendering the "\n" for us, the original README from which they are generated is what we care about.

Either way a possible way for this to look on Linux is:
-e 's/\n/\\\\n\" \+ \n\"/g'
Where you should also add a separate case if the OS is OS X - 
if [ OS == 'OS X']
desc = [sed for OSX]
else
desc = [sed]

Reply:
I'm pushing the Linux change because of backslash rendering on github

Reply:
> On Linux we are also not inserting newlines.... we're inserting "\n" into a C++ string

No, it's replacing an actual newline with a \n **and** a newline, so we keep the formatting of the original Markdown readme file both within the *PluginDescription.h file and in the char array the file returns.

The original (unmodified) sed expression works correctly on both Linux and OS X. I'd be surprised if your escaping of backslashes has broken it too, but would you like to revert your most recent change and retest it on your machine.

Reply:
The below is a snippet from plugins/dummy/DummyPluginDescription.h generated on my machine with the original sed expression by removing that file and then re-running `make builtfiles`. Are you saying this produces different output on your machine? Please can you give us an example?

```
 * This file has been autogenerated by convert_README_to_header.sh, DO NOT EDIT.
 */
#ifndef PLUGINS_DUMMY_DUMMYPLUGINDESCRIPTION_H_
#define PLUGINS_DUMMY_DUMMYPLUGINDESCRIPTION_H_

namespace ola {
namespace plugin {
namespace dummy {

const char plugin_description[] = "Dummy Plugin\n"
"============\n"
"\n"
"The plugin creates a single device with one port. When used as an output\n"
"port it prints the first two bytes of dmx data to stdout.\n"
```

Reply:
On Linux that doesn't work:
`plugins/convert_README_to_header.sh` plugins/dummy/ /tmp/master.out

The above is the version on master, results in (less with line numbers):
![afbeelding](https://user-images.githubusercontent.com/1500246/68545407-a2a71e00-03d5-11ea-8cfa-3abb23951e29.png)

Reply:
I did a push that adds newlines for Linux (but probably not for OS X) however I think we are doing backflips in the air for functionality that is irrelevant.

Reply:
> On Linux that doesn't work:
> `plugins/convert_README_to_header.sh` plugins/dummy/ /tmp/master.out

What you actually mean is it doesn't work on your version of Linux, so what version of sed and what flavour of Linux are you on?

> The above is the version on master

Just to clarify 100%, you mean on the ola master with no other changes? i.e. https://github.com/OpenLightingProject/ola/blob/master/plugins/convert_README_to_header.sh

@FloEdelmann may have some thoughts too.

I'm tempted to say we should probably look at fixing this issue before we get involved in your other change to the sed command too.

Reply:
Ubuntu 19.04

`~$ sed --version
`sed (GNU sed) 4.7

Ran git checkout master which by me is at merge of #1579 then ran the script.

Reply:
As pointed out before this behavior has probably been this way at least since Ubuntu 16.04 (xenial) on which the lint check are run and fail on these files.

Reply:
> As pointed out before this behavior has probably been this way at least since Ubuntu 16.04 (xenial) on which the lint check are run and fail on these files.

The lint checks in master run fine on these files, as per the Travis run against that specific commit:
https://travis-ci.org/OpenLightingProject/ola/jobs/576292203

Unfortunately my bionic change (18.04) is only against 0.10 so far which doesn't have this way of generating the plugin code.

Reply:
Although your earlier bionic run did run, and fail:
https://travis-ci.org/OpenLightingProject/ola/jobs/609809927#L974-L977

```
sh ./plugins/convert_README_to_header.sh ./plugins/artnet ./plugins/artnet/ArtNetPluginDescription.h
sh ./plugins/convert_README_to_header.sh ./plugins/dummy ./plugins/dummy/DummyPluginDescription.h
sed: -e expression #6, char 14: unterminated `s' command
sed: -e expression #6, char 14: unterminated `s' command
```

Is that the same as you get on your personal machine with 19.04?

Reply:
Yup

Reply:
> > As pointed out before this behavior has probably been this way at least since Ubuntu 16.04 (xenial) on which the lint check are run and fail on these files.
> 
> The lint checks in master run fine on these files, as per the Travis run against that specific commit:
> https://travis-ci.org/OpenLightingProject/ola/jobs/576292203
> 
> Unfortunately my bionic change (18.04) is only against 0.10 so far which doesn't have this way of generating the plugin code.

The convert script is new in master? That explains some of it then....

Reply:
> The convert script is new in master? That explains some of it then....

Yeah the convert script is only in master.

I've created a new issue to track some of the issues arising from this https://github.com/OpenLightingProject/ola/issues/1592 .